Data Quality Checks.Rmd

```diff
@@ -202,137 +245,194 @@ scot_check
 
 ```{r, include=FALSE}
 
 #Selecting Council area, HSCP and its localities for latest year in dataset
-#Test case
 #city of edinburgh council = S1200036
 #city of edinburgh partnership = S37000012
 #Edinburgh localities = S99000044,45,46,47
 
-edinburgh_data <- data_indicator %>% 
-  filter(code =="S12000036" | code=="S37000012"|code=="S99000044"|code=="S99000045"|code=="S99000046"|code=="S99000047") %>%
-  mutate(geotype=substr(code,1,3)) %>%
-  group_by(year, geotype) %>%
-  summarise(numerator=sum(numerator)) %>%
-  dcast(year ~ geotype) %>%
-  mutate(ca_partnership=round(S12-S37,2),
-         partnership_localities=round(S99-S37,2),
-         localities_ca=round(S99-S12,2),
-         check_tot=sum(ca_partnership,partnership_localities,localities_ca))
-
-#Filter Scotland value for use in chart
-ed_value <-data_indicator %>% 
+#Filter City of Edinburgh value central line in chart below
+ed_value <- data_indicator %>% 
   filter((code=="S12000036") & year==max(year)) %>% 
   select(rate)
 
+#Generate chart showing rates for all edinburgh geographies for latest year
 edinburgh_chart <- ggplot(data = data_indicator %>% 
          filter((code =="S12000036" | code=="S37000012"
-                |code=="S99000044"|code=="S99000045"|code=="S99000046"|code=="S99000047") & year== max(year)), aes(code, rate)) +
+                |code=="S99000044"|code=="S99000045"|code=="S99000046"|code=="S99000047") 
+                & year== max(year)), aes(areaname, rate)) +
   geom_point(stat = "identity") +
   geom_hline(yintercept= ed_value$rate, linetype="dashed", color = "red")+
   geom_errorbar(aes(ymax=upci, ymin=lowci), width=0.5) +
-  theme(axis.text.x = element_text(angle = 90, hjust = 1))
+  theme(axis.text.x = element_text(angle = 90, hjust = 1))+
+  labs(x = "", y =  "Measure")
+
+# Additional check of numerators (where present) to test if all numerators agree
+
+if(all(data_indicator$numerator != "")){    #if numerator column is not blank
+edinburgh_data <- data_indicator %>%        # create data frame filtered for edinburgh geographies
+  filter(code =="S12000036" | code=="S37000012" | code=="S99000044"| 
+           code=="S99000045"|code=="S99000046"|code=="S99000047") %>%
+  mutate(geotype=substr(code,1,3)) %>%
+  group_by(year, geotype) %>%
+  summarise(numerator=sum(numerator)) %>%
+  spread(geotype, numerator) %>%
+  mutate(ca_partnership=ifelse((hscp==TRUE && ca==TRUE),(round(S12-S37,2)),0),  #create fields that test for differences - use rounded figures
+         partnership_localities=ifelse(locality==TRUE,(round(S99-S37,2)),0), #if dataset includes localities check sum
+         localities_ca=ifelse(locality==TRUE,(round(S99-S12,2)),0),
+         check_tot=round(sum(ca_partnership,partnership_localities,localities_ca),2))}
+
+if(all(data_indicator$numerator == "")){
+  ed_flex <- "No numerator to cross reference totals"
+} else if (sum(edinburgh_data$check_tot) !=0){
```

Comment:
I have added a na.rm=T to the sum as it was causing problems when running the access deprived indicator (as for some years there was no data for some geographies